src/rpc/blockchain.cpp

```diff
@@ -2261,15 +2308,17 @@ static RPCHelpMan scantxoutset()
         int64_t count = 0;
         std::unique_ptr<CCoinsViewCursor> pcursor;
         CBlockIndex* tip;
+        NodeContext& node = EnsureAnyNodeContext(request.context);
         {
+            ChainstateManager& chainman = EnsureChainman(node);
             LOCK(cs_main);
-            ::ChainstateActive().ForceFlushStateToDisk();
-            pcursor = std::unique_ptr<CCoinsViewCursor>(::ChainstateActive().CoinsDB().Cursor());
+            CChainState& active_chainstate = chainman.ActiveChainstate();
+            active_chainstate.ForceFlushStateToDisk();
+            pcursor = std::unique_ptr<CCoinsViewCursor>(active_chainstate.CoinsDB().Cursor());
```

Comment:
Shouldn't Cursor return a unique_ptr?

Reply:
Yes, you're right. `Cursor()` is passing ownership to the caller.

Reply:
I'm going to mark this comment as resolved since this PR shouldn't make any changes here. I think `Cursor()`'s return type should be changed in a separate PR.

Reply:
(unresolved comment to make it easier to see all that needs to be addressed in follow-ups)